### PR TITLE
Enforce and document snapshot limitations regarding vsock devices

### PR DESCRIPTION
**Pull request by acatangiu**
## Reason for This PR

Clarify current support of snapshots on uVMs with Vsock.

## Description of Changes

Add documentation detailing the known limitations with snapshots on uVMs that have a `vsock` device.
Add validation code (+ test) that disallows `diff` snapshots on uVMs that have a `vsock` device.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] ~Any newly added `unsafe` code is properly documented.~
- [x] ~Any API changes are reflected in `firecracker/swagger.yaml`.~
- [x] ~Any user-facing changes are mentioned in `CHANGELOG.md`.~
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: src/vmm/src/rpc_interface.rs [hunk 1]
```diff
@@ -561,6 +566,15 @@ impl RuntimeApiController {
     }
 
     fn create_snapshot(&mut self, create_params: &CreateSnapshotParams) -> ActionResult {
+        // Diff snapshots are not allowed on uVMs with vsock device.
+        if create_params.snapshot_type == SnapshotType::Diff
+            && self.vm_resources.vsock.get().is_some()
+        {
+            return Err(VmmActionError::NotSupported(
+                "Diff snapshots are not allowed on uVMs with vsock device.".to_string(),
```
Comment (sandreim) [inline src/vmm/src/rpc_interface.rs:574]: Nit: this makes sense in a separate PR, or just rename this to account for it:)

File: docs/snapshotting/snapshot-support.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
 # Firecracker Snapshotting
 
+## Table of Contents
```
Comment (sandreim) [inline docs/snapshotting/snapshot-support.md:3]: 👍

**Review by sandreim — APPROVED** [review]
Looks good overall.

**Review by berciuliviu — APPROVED** [review]
LGTM

**Pull request merged**